src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -73,6 +79,16 @@ public function normalize(mixed $object, string $format = null, array $context =
             $object = $object->setTimezone($timezone);
         }
 
+        if ($context[self::TIMESTAMP_CAST_KEY] ?? $this->defaultContext[self::TIMESTAMP_CAST_KEY] ?? false) {
+            return match ($dateTimeFormat) {
+                'U' => $object->getTimestamp(),
+                'U.u' => (float) $object->format('U.u'),
+                'Uv' => (int) $object->format('Uv'),
+                'Uu' => (int) $object->format('Uu'),
```

Comment:
I find those values super obscure, I must admit. Knowing where they're coming from, they might even create the impression that we accept any format modifier here which is clearly not the case.

Maybe an enum would be the better choice?

Reply:
Agreed, or maybe we can leverage private constants here?

Reply:
I don't see any benefits, but well you decide and I'll update

Reply:
On second thought, forget my enum suggestion. I think, my main problem is the guesswork that is encoded in this match block. We basically add an option to cast the result, but we don't say to what we want to cast, so our code has to guess.

What if my app decides to opt in to casting and uses `'Y'` as formatter. From a user perspective, wouldn't I expect an integer cast here as well? I think, we will never be able to completely enumerate all the cases in which a cast makes sense.

So, maybe instead of making the cast option a boolean, we rather accept a type identifier instead?

```php
$this->normalizer->normalize($date, 'json', [
    DateTimeNormalizer::CAST_KEY => 'int',
    DateTimeNormalizer::FORMAT_KEY => 'Y',
]);
```

This would eliminate the need for this guesswork, but it would also mean that the type cast will always remain opt-in.

Reply:
Given that the new option suggests about **timestamp casting only**, I'm not sure why you'd decide to try in your app same thing for the `Y` format. But okay, I see your point, that this would broaden the scope of this feature and allow people for greater usages, so works for me.